cdap-ui/app/cdap/components/AbstractWidget/FormInputs/MultiSelect/index.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+
+interface IMultiSelectProps {
+  value: string | string[];
+  delimiter: string;
+  onChange: (value: string) => void;
+  options: IOption[];
+}
+
+interface IOption {
+  id: string;
+  label: string;
+}
+
+export default function MultiSelect({ value, delimiter, options, onChange }: IMultiSelectProps) {
+  const initSelection = typeof value === 'string' ? value.split(delimiter) : value;
+  const [selections, setSelections] = useState<string[]>(initSelection);
+
+  //  onChangeHandler takes array, turns it into string w/delimiter, and calls onChange on the string
+  const onChangeHandler = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const values = event.target.value as any; // it's expecting a string but multiple select returns an array
+    const selectionsString = values.join(delimiter);
+    setSelections(values);
+    onChange(selectionsString);
+  };
+
+  const optionValues = options.map((opt: IOption) => {
+    return typeof opt === 'string' ? opt : opt.label;
```

Comment:
this doesn't seem to match up the propTypes defined on line 64. Perhaps define the propTypes with PropTypes.oneOf?

Reply:
Good call. It should always be IOption so I fixed it.